### DISCUSSION TITLE
Negative mean for capacitance initialization

### DISCUSSION THREAD

**Discussion by kapilVetrivel** [wkirgsn/thermal-nn#9] [General]
Hello!

Why is the capacitance initialized with negative values?
While I try them with positive values, the loss becomes "nan"

`nn.init.normal_(self.caps, mean=-9.2, std=0.5)  # hand-picked init mean, might be application-dependent`

Thanks

**Comment by wkirgsn** [comment]
later in the forward function of the NN module you will find that `self.caps` is exponentiated.

e^caps

This will always yield positive thermal inverse capacitances. Using a positive value will result in very large numbers that could lead to NaNs.

**Reply by kapilVetrivel** [reply]
Understood & Thanks !

So, if I'm familiar with the thermal network & analytically compute their capacitances, I could as well go ahead & manually initialize the thermal capacitance for each node. Can I expect it to still give a similar prediction?

**Reply by wkirgsn** [reply]
Right, you can initialize the thermal capacitances with your prior knowledge and that should give you a better starting point for the gradient descent training.
Consistently better optima during repeated training are to be expected then.
